string topic_name
float64 rate

